Miscellaneous Cleanup

rnzsgh opened the pull request:
The following changed, primarily in deployment-pipeline.yaml:

- Upgraded the CodeBuild container to the latest Docker version
- Added flags to the ECR login to support the latest docker
- Added NoEcho on GitHub credentials to avoid leaking secrets
- Moved JSON policy documents to YAML
- Simplified configuration in CodeBuild project
- Removed unnecessary quotation marks

I tested the changes in:

- eu-west-1
- us-east-1
- ca-central-1

jpignata commented on templates/deployment-pipeline.yaml:
```diff
@@ -199,7 +199,6 @@ Resources:
                 - Name: BuildOutput
               RunOrder: 1
 
-
```
The convention in these templates is two lines between main stanzas.

jpignata reviewed (changes requested):
LGTM - thank you! Before I merge, please amend your commit message to better describe the changes made (e.g. Miscellaneous Cleanup) and include your bulleted list for posterity. I requested one minor change.

rnzsgh committed:
Misc Cleanup
- Upgraded the CodeBuild container to the latest Docker version
- Added flags to the ECR login to support the latest docker
- Added NoEcho on GitHub credentials to avoid leaking secrets
- Moved JSON policy documents to YAML
- Simplified configuration in CodeBuild project
- Removed unnecessary quotation marks
